content/docs/04-clusters/03-edge/04-edgeforge-workflow/03.5-palette-canvos.md

```diff
@@ -0,0 +1,378 @@
+  stylus:
+    site:
+      edgeHostToken: aUAxxxxxxxxx0ChYCrO
+  install:
+    poweroff: true
+  users:
+    - name: kairos
+      passwd: kairos
+  EOF
+  ```
+
+  </PointsOfInterest>
+  <br />
+
+
+8. Issue the following command to execute the **earthly.sh** file to build the Edge artifacts. The `--PE_VERSION` option in the command below signifies the Palette Edge version to use. The current example uses the **v3.3.3** git tag.
+  <br />
+
+  ```shell
+  ./earthly.sh +build-all-images --PE_VERSION=$(git describe --abbrev=0 --tags)
+  ```
+
+  ```bash coloredLines=2-2|#666666
+  # Output condensed for readability
+  ===================== Earthly Build SUCCESS ===================== 
+  Share your logs with an Earthly account (experimental)! Register for one at https://ci.earthly.dev.
+  ```
+  This command will take up to 15-20 minutes to finish.
+  <br />
+
+</Tabs.TabPane>
+
+<Tabs.TabPane tab="Advanced" key="advanced_create_artifacts">
+
+1. Check out the [CanvOS](https://github.com/spectrocloud/CanvOS.git) GitHub repository containing the starter code. 
+  <br />
+
+  ```bash
+  git clone https://github.com/spectrocloud/CanvOS.git
+  ```
+  
+
+2. Change to the **CanvOS/** directory. 
+  <br />
+
+  ```bash
+  cd CanvOS
+  ```
+
+
+3. The files relevant to the current guide are in the **v3.3.3** tag. Check out the desired tag. You can check out a more recent and stable version if available.
+  <br />
+
+  ```shell
+  git checkout v3.3.3
+  ```
+
+
+4. Review the relevant files.  
+  <br />
+
+  ```bash
+  .
+  ├── .arg          # Defines mutiple variables
+  ├── Dockerfile    # Bakes the arguments to use in the image
+  ├── Earthfile     # Series of commands for creating target images
+  ├── earthly.sh    # Script to invoke the Earthfile, and generate target images
+  └── user-data.template  # A sample user-data file
+  ```
+  <br />
+
+
+5. This step is optional. In this step, you can edit the  **.arg** file to define arguments, such as, custom image tag for provider OS images, image registry, image repository, OS distribution, Kubernetes distribution, and ISO file name. 
+
+  Skip this step if you use the following default values. 
+
+  |**Argument**|**Description**|**Default Value**| **Other Possible Values** |
+  |---|---|---|---|
+  |`CUSTOM_TAG`|Tag for provider image|demo|Lowercase alphanumeric string without spaces|
+  |`IMAGE_REGISTRY`|Image registry name|ttl.sh|Your image registry URL, without http or https <br /> Example: hub.docker.com|
+  |`OS_DISTRIBUTION`|OS Distribution |ubuntu | opensuse-leap|
+  |`IMAGE_REPO`|Image repository name|`$OS_DISTRIBUTION`|Your image repository name|
+  |`OS_VERSION`|OS Version, only applies to Ubuntu |22| 20|
+  |`K8S_DISTRIBUTION`|Kubernetes Distribution |k3s| rke2, kubeadm |
+  |`ISO_NAME`|Name of the Installer ISO|palette-edge-installer|String without spaces|
+
+  Using the default value for all arguments, the provider image name will be similar to `ttl.sh/ubuntu:k3s-1.25.2-v3.4.1-demo`.
```

Comment:
Don't use the defaults. Change it up so the users learn how to play with these values and to provide greater value than the basic section.